### ISSUE TITLE
Mel Filterbank output length difference between speechbrain and torchaudio 

### ISSUE THREAD

**Issue by mzboito** [speechbrain/speechbrain#1179]
Hello. I'm having a size mismatch between the tensors generated by speechbrain's Fbank class and the torchaudio.compliance.kaldi.fbank function. The speechbrain version is always 2 or 3 timesteps longer than the kaldi output, and I have no idea why. I'm attaching some examples of output and the code used to generate them.

**Example:** 
Input: 3 audio files with the following number of samples: 224800, 121632, 143440 (16kHz, mono, wav)

**SB output:**
[1, 1406, 80]
[1, 761, 80]
[1, 897, 80]

**torchaudio output:**
[1, 1403, 80]
[1, 758, 80]
[1, 895, 80]

**###### AUDIO LOADING AND SETTINGS**
import speechbrain as sb
sig = sb.dataio.dataio.read_audio(wav_path)
sig = sig.unsqueeze(0)
sample_rate = 16000
n_mels = 80

**###### SB EXTRACTION**
features_extractor = sb.lobes.features.Fbank(sample_rate=sample_rate, n_mels=n_mels) 
sb_feats = features_extractor(sig)

**###### TORCHAUDIO EXTRACTION**
import torchaudio.compliance.kaldi as ta_kaldi
f_feats = ta_kaldi.fbank(sig, num_mel_bins=n_mels, sample_frequency=sample_rate)
f_feats = f_feats.unsqueeze(0)

**###### VERSION INFORMATION**
python=3.8.12
torch=1.10.0+cu102
numpy=1.19.2
speechbrain=0.5.10
torchaudio=0.10.0+cu102

Thanks a lot for your help!

**Comment by mzboito** [comment]
Hello Gastron. Yes, it might be that the default settings for this function on speechbrain are slightly different than the torchaudio
version. However, I did check kaldi implementation, and I didn't find the root of this difference. https://pytorch.org/audio/stable/compliance.kaldi.html#fbank
https://pytorch.org/audio/stable/_modules/torchaudio/compliance/kaldi.html

I don't really need my features to be of the same exact size, I would just like to be sure that the features from both are equivalent, and that I can compare models using features extracted from different libraries.

**Comment by Gastron** [comment]
@mravanelli might have some idea about where this difference in number of frames comes from.

Anyway, as long as the feature parameters such as number of Mel-scale filters etc. are the same, I would claim the features are comparable - both are fine. However I would hesitate to apply a model trained on Kaldi features on features from the SpeechBrain side or vice versa - probably that would work fine, but in that case I would prefer to just use the matching feature extractor.

**Comment by mravanelli** [comment]
Yes, the difference is due to how padding is managed. Maybe in the future
we can add an option to allow users to set the desired padding option.
